### ISSUE TITLE
Handle errors in the viewmodels

### ISSUE THREAD

**Issue by mchac15** [HikeMate/hikeMateApp#132]
In the current state of the app, errors are often silenced in the view models when they occur in the model. Errors should be handled in the UI.

**Definition of Done**

- [x] Identify where errors are silenced by view models
- [ ] Do one PR for each view model that silences errors and actually handle the error in the UI appropriately, so the user gets appropriate and understandable feedback
   - [ ] `AuthViewModel`
   - [ ] ~~`SavedHikesViewModel`~~
   - [ ] ~~`ListOfHikeRoutesViewModel`~~
   - [ ] `HikesViewModel`

**Comment by MaelImhof** [comment]
- The profile view model will already be modified as part of another task, no need to touch it for this task
- The auth view model has a few problems of its own, errors need to be handled. However, modifications are still ongoing for this view model, I should fix it once they are done.
- Saved hikes view model was already designed to handle errors in the UI, but a quick double-check wouldn't hurt
- Same for list of hike routes view model, it was already modified to handle errors to some extent, see if we can make it better

**Comment by Wylwi** [comment]
@MaelImhof can this issue be closed?

**Comment by MaelImhof** [comment]
@Wylwi not now, I only did one screen out of three. I'm actually waiting on @mchac15 and/or @davidholzwarth for their job relative to details screen, I don't know where they're at currently.

**Comment by MaelImhof** [comment]
I'm working on better error handling in the `AuthViewModel`, however since #187 exists now, the two others don't make much sense anymore. Once I merge all hike view models into a single one, I'll do proper error handling in there as well.

**Comment by rBergCode** [comment]
I will close this as I created 2 new tasks "Handle errors in new HikesViewModel" and "Handle errors in AuthViewModel"
